### ISSUE TITLE
[linux] prey user is not added to required groups on non-debian distros

### ISSUE THREAD

**Issue by mattlyons0** [prey/prey-node-client#65]
EDIT: This started as a completely different issue and I slowly resolved it by traversing the source code. Now the only issue is using the webcam. 

Prey fails to map its port and seems to have issues connecting as a result of that.

    PREY 1.3.5 spreads its wings!
  Current time: Thu Jan 08 2015 08:55:50 GMT-0600 (CST)
  Running with PID 2721 as prey over Node.js v0.10.35 on a x64 linux system (3.17.6-1-ARCH)

     info Thu, 08 Jan 2015 14:55:50 GMT [triggers] Watching: connection, network, power
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Starting trigger: connection
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Running: connection
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Starting trigger: network
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Starting trigger: power
     info Thu, 08 Jan 2015 14:55:50 GMT [control-panel] Queueing check-ins every 20 minutes.
   notice Thu, 08 Jan 2015 14:55:50 GMT [control-panel] Checking for existing port mappings...
     info Thu, 08 Jan 2015 14:55:50 GMT Plugin loaded: control-panel
     info Thu, 08 Jan 2015 14:55:50 GMT 1 plugins loaded with 0 errors.
     info Thu, 08 Jan 2015 14:55:50 GMT Initialized.
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Running: network
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Running: power
     warn Thu, 08 Jan 2015 14:55:50 GMT Relaunching 2 commands previously in execution.
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Starting action: alert
     info Thu, 08 Jan 2015 14:55:50 GMT [actions] Starting action: lock
     warn Thu, 08 Jan 2015 14:55:51 GMT [control-panel] Unable to sync settings.
     info Thu, 08 Jan 2015 14:55:55 GMT [control-panel] No previously mapped ports found.
     info Thu, 08 Jan 2015 14:55:55 GMT [control-panel] Attempting to map external port 30334
     warn Thu, 08 Jan 2015 14:55:56 GMT [control-panel] Got 1 commands.
     info Thu, 08 Jan 2015 14:56:13 GMT [reports] Gathering status report.
    error Thu, 08 Jan 2015 14:56:13 GMT [reports] Unable to get logged_user: No logged user detected.
    error Thu, 08 Jan 2015 14:56:31 GMT [control-panel] Unable to map port: Could not locate gateway on time.
     info Thu, 08 Jan 2015 14:56:31 GMT [control-panel] Queueing check-ins every 0.5 minutes.
     info Thu, 08 Jan 2015 14:56:32 GMT [reports] Report gathered: status
     info Thu, 08 Jan 2015 14:56:32 GMT [control-panel] Posting data
     info Thu, 08 Jan 2015 14:56:32 GMT [control-panel] Got error: connect ECONNREFUSED
    error Thu, 08 Jan 2015 14:57:02 GMT Error: connect ECONNREFUSED
    error Thu, 08 Jan 2015 14:57:31 GMT Error: connect ECONNREFUSED
    error Thu, 08 Jan 2015 15:00:31 GMT Error: connect ECONNREFUSED
     warn Thu, 08 Jan 2015 15:01:18 GMT [control-panel] Got 1 commands.
     info Thu, 08 Jan 2015 15:01:33 GMT [reports] Gathering status report.
    error Thu, 08 Jan 2015 15:01:33 GMT [reports] Unable to get logged_user: No logged user detected.
     info Thu, 08 Jan 2015 15:01:47 GMT [reports] Report gathered: status
     info Thu, 08 Jan 2015 15:01:47 GMT [control-panel] Posting data
     info Thu, 08 Jan 2015 15:02:25 GMT [control-panel] Got status 200 from server: OK

I am on wifi using Connman as my network manager. 
ip link:

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: wlp2s0: <BROADCAST,MULTICAST,DYNAMIC,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DORMANT group default qlen 1000
    link/ether e8:2a:ea:7d:da:16 brd ff:ff:ff:ff:ff:ff

journalctl -b | grep prey
Jan 08 08:55:50 ArchBook sudo[2770]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:55:53 ArchBook sudo[2786]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:55:56 ArchBook sudo[2808]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:55:56 ArchBook sudo[2817]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:55:59 ArchBook sudo[2830]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:56:02 ArchBook sudo[2843]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:56:13 ArchBook sudo[2907]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan
Jan 08 08:56:21 ArchBook sudo[2957]: prey : TTY=unknown ; PWD=/opt/prey-node-client ; USER=root ; COMMAND=/usr/sbin/iwlist wlp2s0 scan

The unable to get logged user error is because I didn't have openssh installed (which provides the ssh-agent command).

If you need any more information feel free to ask.

**Comment by mauriciooo** [comment]
Hello @mattlyons0, thanks for your report.

Even though `[control-panel] Unable to map port: Could not locate gateway on time.` is stated as an error, it's more of a warning. It means Prey was unable to use upnp to connect with our servers. When this happens, Prey changes to interval mode, meaning that every certain amount of time (30 seconds by default) it will try to retrieve instructions from our [website](https://panel.preyproject.com).

Now, the message
```
info Thu, 08 Jan 2015 14:56:32 GMT [control-panel] Posting data
info Thu, 08 Jan 2015 14:56:32 GMT [control-panel] Got error: connect ECONNREFUSED
```
it's a different issue. It means for some reason the client was unable to post data to our servers. 

Could you please paste the content of the config file (**please make sure to leave out your api_key and device_key**) located in `/etc/prey/prey.conf`?.

**Comment by mattlyons0** [comment]
Actually the error with the gateway was due to not being able to locate a user since I didnt have the ssh client installed. Currently the only issue is prey using the webcam when the device is missing. I can't figure out the exact command that is being issued to streamer but the error is the following in the log:

    error Thu, 08 Jan 2015 15:41:30 GMT [reports] Unable to get picture: Command failed:$
v4l2: open /dev/video0: Permission denied
vid-open: failed: libv4l
no grabber device available

What is the extension being passed to call_streamer? I can't seem to be able to find the call.
https://github.com/prey/prey-node-client/blob/a7366cc1a3975a7c684fdd41415f7ee5577473d0/lib/agent/providers/webcam/linux/index.js

**Comment by mauriciooo** [comment]
The extension being passed to the call_streamer function is `.jpeg`.

`v4l2: open /dev/video0: Permission denied`: I have the suspicion that the `prey` user might not have the required permissions to use video. Could you please check if the `prey` user has access to the `video` group?

Also, are you running Arch Linux?

**Comment by mattlyons0** [comment]
Yes. That is actually what I was just looking into. I was wondering if xawtv added its own group but then I stumbled upon the video group.

**Comment by mauriciooo** [comment]
@mattlyons0 Thanks for the feedback. The permissions issue might be related with the PR #50. Is adduser available in Arch?

**Comment by mattlyons0** [comment]
Nope, "gpasswd --add USERNAME GROUP" is how you add groups to a user.

**Comment by lemavri** [comment]
Thanks. We are already thinking of using any of the options provided by the `shadow` package such as `gpasswd` or `usermod` to make sure Prey works with the rest of the distros that are not debian-based.
